include/swift/RemoteInspection/TypeRefBuilder.h

```diff
@@ -567,62 +567,58 @@ class TypeRefBuilder {
     return nullptr;
   }
 
+  // Recursively constructs TypeRef for a bound generic, including
+  // the enclosing (parent) generic contexts
```

Comment:
I've significantly rearranged the logic here to make it more cleanly recursive (which allowed me to eliminate the iterative walk to find a parent type).